.github/workflows/coverage.yml

```diff
@@ -0,0 +1,71 @@
+name: Coverage
+
+on:
+  push:
+    branches:
+      - master
+  pull_request: {}
+
+jobs:
+
+  coverage:
+    runs-on: ubuntu-latest
+
+    # Not intended for forks.
+    if: github.repository == 'optuna/optuna'
+
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup Python3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.7
+
+    - name: Setup cache
+      uses: actions/cache@v2
+      env:
+        cache-name: coverage
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}-v1
+        restore-keys: |
+          ${{ runner.os }}-${{ env.cache-name }}-${{ hashFiles('**/setup.py') }}
+
+    - name: Setup environment
+      run: |
+        sudo apt-get update
+        sudo apt-get -y install openmpi-bin libopenmpi-dev
+
+    - name: Install
+      run: |
+        python -m pip install --upgrade pip
+        pip install --progress-bar off -U setuptools
+
+        # Install minimal dependencies and confirm that `import optuna` is successful.
+        pip install --progress-bar off .
+        python -c 'import optuna'
+        optuna --version
+
+        pip install --progress-bar off .[tests]
+        pip install --progress-bar off .[optional]
+        pip install --progress-bar off .[integration] -f https://download.pytorch.org/whl/torch_stable.html
+        pip install --progress-bar off .[codecov]
+
+    - name: Tests
+      env:
+        OMP_NUM_THREADS: 1
+      run: |
+        pytest --cov=optuna --cov-report=xml tests
+
+    - name: Multi-node tests
+      env:
+        OMP_NUM_THREADS: 1
+      run: |
+        mpirun -n 2 -- pytest tests/integration_tests/test_chainermn.py
```

Comment:
Probably outside the scope of this PR but I noticed that this pytest execution isn't recording the coverage?

Reply:
Okay, then remove this. I copied things from Circle CI config, sorry.

Reply:
Created an issue at https://github.com/optuna/optuna/issues/2038.